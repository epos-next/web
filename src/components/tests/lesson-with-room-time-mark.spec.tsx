import LessonWithRoomTimeMark from "@components/lesson-with-room-time-mark";
import { render } from "@testing-library/react";
import React from "react";

const props = {
    mark: 5,
    room: "202",
    time: "11:00 - 11:45",
    subject: "Алгебра"
}

it("should match snapshot", () => {
    const { asFragment } = render(<LessonWithRoomTimeMark {...props} />)
    expect(asFragment()).toMatchSnapshot()
})

it("should use mark prop", () => {
    const { getByTestId, rerender } = render(<LessonWithRoomTimeMark {...props} mark={5} />)
    expect(getByTestId("mark")).toHaveTextContent("5")

    rerender(<LessonWithRoomTimeMark {...props} mark={4} />)
    expect(getByTestId("mark")).toHaveTextContent("4")

    rerender(<LessonWithRoomTimeMark {...props} mark={3} />)
    expect(getByTestId("mark")).toHaveTextContent("3")

    rerender(<LessonWithRoomTimeMark {...props} mark={2} />)
    expect(getByTestId("mark")).toHaveTextContent("2")

    rerender(<LessonWithRoomTimeMark {...props} mark={1} />)
    expect(getByTestId("mark")).toHaveTextContent("1")
});

it("should use room prop", () => {
    const { getByTestId, rerender } = render(<LessonWithRoomTimeMark {...props} />)
    expect(getByTestId("lesson-subtitle")).toHaveTextContent("202")

    rerender(<LessonWithRoomTimeMark {...props} room="101"/>)
    expect(getByTestId("lesson-subtitle")).toHaveTextContent("101")

    rerender(<LessonWithRoomTimeMark {...props} room="1234567890"/>)
    expect(getByTestId("lesson-subtitle")).toHaveTextContent("1234567890")
})

it("should use time prop", () => {
    const { getByTestId, rerender } = render(<LessonWithRoomTimeMark {...props} time="11:00 - 12:00" />)
    expect(getByTestId("lesson-subtitle")).toHaveTextContent("11:00 - 12:00")

    rerender(<LessonWithRoomTimeMark {...props} time="9:00 - 15:00"/>)
    expect(getByTestId("lesson-subtitle")).toHaveTextContent("9:00 - 15:00")

    rerender(<LessonWithRoomTimeMark {...props} time="valabuday"/>)
    expect(getByTestId("lesson-subtitle")).toHaveTextContent("valabuday")
})

it("should remove class from inner lesson", () => {
    const { getByTestId } = render(<LessonWithRoomTimeMark {...props} />)
    expect(getByTestId("lesson-container")).not.toHaveAttribute("className")
});

