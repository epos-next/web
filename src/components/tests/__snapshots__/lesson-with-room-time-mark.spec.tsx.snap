// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  .c5 {
  font-size: 17px;
  line-height: 20px;
  color: var(--primary);
}

.c6 {
  font-size: 15px;
  line-height: 18px;
  color: var(--secondary);
}

.c3 {
  width: 36px;
  height: 36px;
  border-radius: 50%;
  background-color: #F8CBC4;
  color: #F18477;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  font-weight: 500;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-left: 15px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  overflow: hidden;
}

.c8 {
  width: 3px;
  height: 3px;
  background: #696969;
  border-radius: 50%;
  margin: 0 10px;
}

.c9 {
  font-size: 15px;
  line-height: 18px;
  font-weight: 500;
  color: var(--primary);
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

<div
    class="c0 lesson-with-room-time-mark"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
        data-testid="lesson-container"
      >
        <div
          class="c3"
          color="#F8CBC4"
          data-testid="avatar"
        >
          А
        </div>
        <div
          class="c4"
        >
          <span
            class="c5"
            data-testid="lesson-name"
          >
            Алгебра
          </span>
          <span
            class="c6"
            data-testid="lesson-subtitle"
          >
            <div
              class="c7"
            >
              202
              <div
                class="c8"
              />
              11:00 - 11:45
            </div>
          </span>
        </div>
      </div>
    </div>
    <span
      class="c9"
      data-testid="mark"
    >
      5
    </span>
  </div>
</DocumentFragment>
`;
